File: 21-merge-two-sorted-lists/merge-two-sorted-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
 typedef struct ListNode * NODE;
 NODE getnode(){
    return (NODE)malloc(sizeof(struct ListNode));
 }
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2) {
    NODE  l3=NULL,temp1=l1,temp2=l2;
    NODE end=NULL;
    while (temp1!=NULL && temp2!=NULL){
        
        NODE newnode=getnode();
        if (temp1->val > temp2->val){
            newnode->val=temp2->val;
            temp2=temp2->next;
        }
        else if (temp1->val <= temp2->val){
            newnode->val=temp1->val;
            temp1=temp1->next;
        }
        newnode->next=NULL;
        if (end==NULL){
            l3=newnode;}
        else {end->next=newnode;}
        end=newnode;
    }
    NODE temp=(temp1==NULL)?temp2:temp1;
    while (temp!=NULL){
        NODE newnode=getnode();
        newnode->val=temp->val;
        newnode->next=NULL;
        if (end==NULL){
            l3=newnode;}
        else{
        end->next=newnode;}
        end=newnode;
        temp=temp->next;
    }
    return l3;
}

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        start=None
        prev_node=None
        while list1 and list2:
            cur_node=ListNode()
            if start==None:
                start=cur_node
            if list1.val>list2.val:
                cur_node.val=list2.val
                list2=list2.next
                
            else:
                cur_node.val=list1.val
                list1=list1.next
            if prev_node:
                prev_node.next=cur_node
            prev_node=cur_node
        if list1:
            if prev_node:
                prev_node.next=list1
            else:
                return list1
        elif list2:
            if prev_node:
                prev_node.next=list2
            else:
                return list2
        return start
